packages/plugin-aws-sdk/src/services/sqs.ts

```diff
@@ -45,16 +77,35 @@ export class SqsServiceExtension implements ServiceExtension {
     const operation = (request as any)?.operation;
     switch (operation) {
       case "receiveMessage":
-        isIncoming = true;
-        spanKind = SpanKind.CONSUMER;
-        spanName = queueName;
-        spanAttributes[SqsAttributeNames.MESSAGING_OPERATION] = "receive";
+        {
+          isIncoming = true;
+          spanKind = SpanKind.CONSUMER;
+          spanName = queueName;
+          spanAttributes[SqsAttributeNames.MESSAGING_OPERATION] = "receive";
+
+          const params: Record<string, any> = (request as any).params;
+          const attributesNames = params.MessageAttributeNames || [];
+          attributesNames.push(TRACE_PARENT_HEADER);
+          params.MessageAttributeNames = attributesNames;
+        }
         break;
 
       case "sendMessage":
       case "sendMessageBatch":
-        spanKind = SpanKind.PRODUCER;
-        spanName = queueName;
+        {
+          spanKind = SpanKind.PRODUCER;
+          spanName = queueName;
+
+          const params: Record<string, any> = (request as any).params;
+          const attributes = params.MessageAttributes || {};
+          if (attributes.length < SQS_MAX_MESSAGE_ATTRIBUTES) {
+            propagation.inject(attributes, contextSetterFunc);
+            params.MessageAttributes = attributes;
+          } else {
+            // TODO: in this case we are not setting the context propagtion for consumers
```

Comment:
I think oteljs has a logger, so let's just log a message saying that we can't add context propagation attribute. WDYT?

Reply:
Good point. Added logging